routes/model/model.js

```diff
@@ -49,6 +53,12 @@ class RouteModel {
             });
         }
 
+        // Only admin user can create, update or delete employee or product,
+        // If this model is employee or product, set middleware to authenticate is
+        // this user admin
+        if (REQUIRE_ADMIN.includes(this.model_name))
+            this.model_router.use(authenticate_admin);
```

Comment:
Follow the comment above to split up using the authenticate_admin middleware for each default model method.

Example:
if (!this.disabled_routes.includes(RESTAPI_TYPES.POST)) {

            // Add conditional statement here to check authenticate_admin middleware is required
            // If conditional statement is true, add the middleware to the post method

            this.model_router.post('/', async (req, res) => {
                try {
                    const insert_payload = { ...req.body };
                    await knex(this.model_name).insert(insert_payload);
                    res.status(201).json({ message: `Create ${this.model_name} successfully!` });
                } catch (err) {
                    res.status(400).json({ message: err.message });
                }
            });
        }